src/heightmap.cpp

```diff
@@ -255,8 +295,8 @@ static bool ReadHeightmapBMP(const char *filename, uint *x, uint *y, byte **map)
 		return false;
 	}
 
-	/* Check if image dimensions don't overflow a size_t to avoid memory corruption. */
-	if ((uint64)info.width * info.height >= (size_t)-1 / (info.bpp == 24 ? 3 : 1)) {
+	/* 1 byte per pixel for the map, and 3 or 1 bytes per pixel for loading the BMP data. */
```

Comment:
What do you mean "for the map"?

Reply:
The `byte **map` parameter of this function that is MallocT-ed about 20 lines below here.

Reply:
Okay, so this does a bit more than I expected.

Basically, for PNGs you only say: it allocated the "map" only, so as long as that is in bounds, I am fine. You forgive the PNG buffer basically.
For BMPs, you say: we will also allocated based on bpp, so that also has to fit in this value.

Do I get that right?

Reply:
Yeah, mostly because BMP allocation goes via MallocT and thus "crashes" the game. Whereas PNG allocation goes via malloc and just returns a PNG error. I have removed the distinction for number of bytes to allocate though.